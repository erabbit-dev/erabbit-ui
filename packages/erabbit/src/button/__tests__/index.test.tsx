import { Button } from '..'
import { mount } from '@vue/test-utils'

test('should render default button', () => {
  const wrapper = mount(Button, {
    slots: {
      default: 'Submit',
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
})

test('should trigger click event', () => {
  const wrapper = mount(Button, {
    slots: {
      default: 'Submit',
    },
    props: {
      size: 'large',
      type: 'primary',
      icon: 'cart',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('should trigger click event when loading or disabled', async () => {
  const fn = vi.fn()
  const wrapper = mount({
    setup() {
      return () => <Button onClick={() => fn()}>Submit</Button>
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.find('.er-button').trigger('click')
  expect(fn).toBeCalledTimes(1)

  await wrapper.setProps({
    disabled: true,
  })
  await wrapper.find('.er-button').trigger('click')
  expect(fn).toBeCalledTimes(1)

  await wrapper.setProps({
    disabled: false,
  })
  await wrapper.find('.er-button').trigger('click')
  expect(fn).toBeCalledTimes(2)

  await wrapper.setProps({
    loading: true,
  })
  await wrapper.find('.er-button').trigger('click')
  expect(fn).toBeCalledTimes(2)
})
