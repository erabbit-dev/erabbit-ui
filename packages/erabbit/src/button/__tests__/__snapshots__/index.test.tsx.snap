// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render default button 1`] = `
"<button class=\\"er-button ellipsis er-button--middle er-button--default\\">
  <!----><span class=\\"er-button__text\\">Submit</span>
</button>"
`;

exports[`should trigger click event 1`] = `"<button class=\\"er-button ellipsis er-button--large er-button--primary\\"><i class=\\"er-icon er-icon--cart er-button__icon\\" style=\\"font-size: 1em;\\"></i><span class=\\"er-button__text\\" style=\\"padding-left: 5px;\\">Submit</span></button>"`;

exports[`should trigger click event when loading or disabled 1`] = `
"<button class=\\"er-button ellipsis er-button--middle er-button--default\\">
  <!----><span class=\\"er-button__text\\">Submit</span>
</button>"
`;
